File: 2399-check-distances-between-same-letters/2399-check-distances-between-same-letters.py
class Solution:
    def checkDistances(self, s: str, distance: list[int]) -> bool:
        mp = {}

        for i in range(len(s)):
            if s[i] not in mp:
                mp[s[i]] = []
            mp[s[i]].append(i)

        for ch, pos in mp.items():
            diff = pos[1] - pos[0] - 1

            if diff != distance[ord(ch) - ord('a')]:
                return False

        return True

File: 2399-check-distances-between-same-letters/2399-check-distances-between-same-letters.cpp
class Solution {
public:
    bool checkDistances(string s, vector<int>& distance) {
        unordered_map<char,vector<int>> mp;
        int n=s.length();
        for(int i=0;i<n;i++){
            mp[s[i]].push_back(i);
        }
        for(auto &it:mp){
            int diff=it.second[1]-it.second[0]-1;
            if(diff!=distance[it.first-'a'])return false;
        }
        return true;
    }
};